src/QMCWaveFunctions/TrialWaveFunction.cpp

```diff
@@ -136,57 +136,52 @@ TrialWaveFunction::RealType TrialWaveFunction::evaluateLog(ParticleSet& P)
   return LogValue;
 }
 
-void TrialWaveFunction::flex_evaluateLog(const RefVectorWithLeader<TrialWaveFunction>& wf_list,
-                                         const RefVectorWithLeader<ParticleSet>& p_list)
+void TrialWaveFunction::mw_evaluateLog(const RefVectorWithLeader<TrialWaveFunction>& wf_list,
+                                       const RefVectorWithLeader<ParticleSet>& p_list)
 {
-  if (wf_list.size() > 1)
-  {
-    auto& wf_leader = wf_list.getLeader();
-    auto& p_leader  = p_list.getLeader();
-    ScopedTimer local_timer(wf_leader.TWF_timers_[RECOMPUTE_TIMER]);
-
-    constexpr RealType czero(0);
-    const auto g_list(TrialWaveFunction::extractGRefList(wf_list));
-    const auto l_list(TrialWaveFunction::extractLRefList(wf_list));
-
-    int num_particles = p_leader.getTotalNum();
-    auto initGandL    = [num_particles, czero](TrialWaveFunction& twf, ParticleSet::ParticleGradient_t& grad,
-                                            ParticleSet::ParticleLaplacian_t& lapl) {
-      grad.resize(num_particles);
-      lapl.resize(num_particles);
-      grad           = czero;
-      lapl           = czero;
-      twf.LogValue   = czero;
-      twf.PhaseValue = czero;
-    };
-    for (int iw = 0; iw < wf_list.size(); iw++)
-      initGandL(wf_list[iw], g_list[iw], l_list[iw]);
+  auto& wf_leader = wf_list.getLeader();
+  auto& p_leader  = p_list.getLeader();
+  ScopedTimer local_timer(wf_leader.TWF_timers_[RECOMPUTE_TIMER]);
 
-    auto& wavefunction_components = wf_leader.Z;
-    const int num_wfc             = wf_leader.Z.size();
-    for (int i = 0, ii = RECOMPUTE_TIMER; i < num_wfc; ++i, ii += TIMER_SKIP)
-    {
-      ScopedTimer z_timer(wf_leader.WFC_timers_[ii]);
-      const auto wfc_list(extractWFCRefList(wf_list, i));
-      wavefunction_components[i]->mw_evaluateLog(wfc_list, p_list, g_list, l_list);
-      auto accumulateLogAndPhase = [](TrialWaveFunction& twf, WaveFunctionComponent& wfc) {
-        twf.LogValue += std::real(wfc.LogValue);
-        twf.PhaseValue += std::imag(wfc.LogValue);
-      };
-      for (int iw = 0; iw < wf_list.size(); iw++)
-        accumulateLogAndPhase(wf_list[iw], wfc_list[iw]);
-    }
-    auto copyToP = [](ParticleSet& pset, TrialWaveFunction& twf) {
-      pset.G = twf.G;
-      pset.L = twf.L;
+  constexpr RealType czero(0);
+  const auto g_list(TrialWaveFunction::extractGRefList(wf_list));
+  const auto l_list(TrialWaveFunction::extractLRefList(wf_list));
+
+  int num_particles = p_leader.getTotalNum();
+  auto initGandL    = [num_particles, czero](TrialWaveFunction& twf, ParticleSet::ParticleGradient_t& grad,
```

Comment:
Maybe a comment that this mess is due to design issues with Particle Set and Wave Function not because the current PR's idea is complicated.